### PR TITLE
documentation: update first-party algorithms and structural updates

### PR DESCRIPTION
**Pull request by ragdhall**
*Issue #, if available:*

*Description of changes:* Update First-Party Algorithms to Built-In Algorithms. Change structure of first-party algorithms to arrange by data type. Changes are according to the discussion in this doc: https://quip-amazon.com/4LBNADovHMck/SDK-Update#temp:C:PRT6b604a5f6b034bbeabcf72962

*Testing done:*

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I certify that the changes I am introducing will be backword compatible, and I have discussed concerns about this, if any, with the Python SDK team
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [x] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have added unit and/or integration tests as appropriate to ensure backward compatibility of the changes
- [x] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [x] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: doc/algorithms/index.rst [hunk 1]
```diff
@@ -1,20 +1,15 @@
 ######################
-First-Party Algorithms
+Built-in Algorithms
 ######################
```
Comment (bencrabtree) [inline doc/algorithms/index.rst:3]: I'm pretty sure you need the `#`'s to be the same length as the text. That is probably one of the reasons why your codebuild is failing.

**Review by bencrabtree — CHANGES_REQUESTED** [review]
Looks good overall, just make sure that all codebuilds are passing

File: doc/algorithms/index.rst (hunk 1 above)
Comment (ragdhall) [reply doc/algorithms/index.rst:3]: I think it needs to be the same or longer in length. `make html` as well as `tox -e twine,sphinx` pass locally. The build errors seem unrelated to my changes.

Comment (bencrabtree) [reply doc/algorithms/index.rst:3]: sounds good to me

**Pull request merged**